docs/orchestration.asciidoc

```diff
@@ -0,0 +1,104 @@
+[id="{p}-orchestration"]
+=== NodeSets orchestration
+
+[id="{p}-nodesets"]
+==== NodeSets overview
+
+The Elasticsearch cluster is specified using a list of NodeSets. Each NodeSet represents a group of Elasticsearch nodes sharing the same specification (both Elasticsearch configuration and Kubernetes Pod configuration).
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: elasticsearch.k8s.elastic.co/{eck_crd_version}
+kind: Elasticsearch
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  nodeSets:
+  - name: master-nodes
```

Comment:
Nit:
I tend to think this is the name for each node of the set.
I would stay for the name with `master` and `data`, or `master-node` and `data-node` in the singular.
 
```sh
> k get pods
NAME                  READY   STATUS        RESTARTS   AGE
z-es-data-nodes-0     1/1     Running       0          60s
z-es-data-nodes-1     1/1     Running       0          60s
z-es-data-nodes-2     1/1     Running       0          60s
z-es-master-nodes-0   1/1     Running       0          60s
z-es-master-nodes-1   1/1     Running       0          60s
z-es-master-nodes-2   1/1     Running       0          60s
```

Reply:
I think that you are right that pod name that contains plural is weird looking, but I think I'd still go for it. For me, the `name` reflects the name of the set and the fact we use it it such way for pod names is secondary. I wouldn't dictate the name based on the template we use for pods (eg. if we would have `pod-0-in-master-nodes` pod names it would be fine and singular would look weird).

Reply:
My nit would be more just omitting `-nodes` since it doesn't really add any descriptive value, i.e. just use `- name: masters`. But that is very nitpicky

Reply:
`masters` looks very nice to me. But `datas` does not?

With only `master` and `data` I'm afraid people will think "this is the way to create a master node". It's not, it's just a name.

Reply:
>masters looks very nice to me. But datas does not?

Data is already plural (datum singular), so I agree. IMO we don't need to overthink this, `master-nodes` or `masters` doesn't make a huge difference for me. We can add a comment that is along the lines of `name: masters # note this is just an arbitrary name and could also be nickelback-is-underrated-fight-me` to maybe help clear up the potential confusion you mentioned

Reply:
`master-node` and `data-node` (singular)
My two cents.